compose/config/config.py

```diff
@@ -205,10 +205,10 @@ def with_abs_paths(cls, working_dir, filename, name, config):
             config)
 
 
-def find(base_dir, filenames):
+def find(base_dir, filenames, override_dir=None):
     if filenames == ['-']:
         return ConfigDetails(
-            os.getcwd(),
+            override_dir or os.getcwd(),
```

Comment:
I just noticed something which I missed earlier.

We already have `base_dir` here. I think setting `project-directory` should be equivalent to setting `base_dir`. I think we can probably re-use it. Instead of using `os.getcwd()`, this can be `os.abspath(base_dir)`, so when we pass in the default of `.` it will be expanded to the same path.

Sorry I didn't notice this earlier.